src/common/walFilter/walFilter.c

```diff
@@ -0,0 +1,720 @@
+#include "build.auto.h"
+
+#include "walFilter.h"
+
+#include "common/debug.h"
+#include "common/log.h"
+#include "common/type/object.h"
+
+#include "common/compress/helper.h"
+#include "common/partialRestore.h"
+#include "config/config.h"
+#include "postgres/interface/crc32.h"
+#include "postgres/version.h"
+#include "postgresCommon.h"
+#include "storage/helper.h"
+#include "versions/recordProcessGPDB6.h"
+
+typedef enum
+{
+    noStep, // This means that the process of reading the record is in an uninterrupted state.
+    stepBeginOfRecord,
+    stepReadHeader,
+    stepReadBody,
+} ReadStep;
+
+typedef struct WalInterface
+{
+    unsigned int pgVersion;
+    StringId fork;
+
+    uint16_t header_magic;
+    void (*validXLogRecordHeader)(const XLogRecord *record, PgPageSize heapPageSize);
+    void (*validXLogRecord)(const XLogRecord *record, PgPageSize heapPageSize);
+} WalInterface;
+
+static WalInterface interfaces[] = {
+    {PG_VERSION_94, CFGOPTVAL_FORK_GPDB, GPDB6_XLOG_PAGE_MAGIC, validXLogRecordHeaderGPDB6, validXLogRecordGPDB6}
+};
+
+typedef struct WalFilter
+{
+    ReadStep currentStep;
+    const unsigned char *page;
+
+    PgPageSize heapPageSize;
```

Comment:
There is a mixture of styles in fields naming - both snake_case and camelCase. Lets make it consistent.